bundles/org.openhab.binding.myq/src/main/java/org/openhab/binding/myq/internal/handler/MyQAccountHandler.java

```diff
@@ -0,0 +1,352 @@
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.FieldNamingPolicy;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+/**
+ * The {@link MyQAccountHandler} is responsible for communicating with the MyQ API based on an account.
+ *
+ * @author Dan Cunningham - Initial contribution
+ */
+@NonNullByDefault
+public class MyQAccountHandler extends BaseBridgeHandler {
+    private static final String BASE_URL = "https://api.myqdevice.com/api";
+    private static final Integer RAPID_REFRESH_SECONDS = 5;
+    private final Logger logger = LoggerFactory.getLogger(MyQAccountHandler.class);
+    private final Gson gsonUpperCase = new GsonBuilder().setFieldNamingPolicy(FieldNamingPolicy.UPPER_CAMEL_CASE)
+            .create();
+    private final Gson gsonLowerCase = new GsonBuilder()
+            .setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES).create();
+    private @Nullable Future<?> normalPollFuture;
+    private @Nullable Future<?> rapidPollFuture;
+    private @Nullable String securityToken;
+    private @Nullable AccountDTO account;
+    private @Nullable DevicesDTO devicesCache;
+    private Integer normalRefreshSeconds = 60;
+    private HttpClient httpClient;
+    private String username = "";
+    private String password = "";
+    private String userAgent = "";
+
+    public MyQAccountHandler(Bridge bridge, HttpClient httpClient) {
+        super(bridge);
+        this.httpClient = httpClient;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    @Override
+    public void initialize() {
+        MyQAccountConfiguration config = getConfigAs(MyQAccountConfiguration.class);
+        normalRefreshSeconds = config.refreshInterval;
+        username = config.username;
+        password = config.password;
+        // MyQ can get picky about blocking user agents apparently
+        userAgent = MyQAccountHandler.randomString(40);
+        securityToken = null;
+        updateStatus(ThingStatus.UNKNOWN);
+        restartPolls(false);
+    }
+
+    @Override
+    public void dispose() {
+        stopPolls();
+    }
+
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return Collections.singleton(MyQDiscoveryService.class);
+    }
+
+    @Override
+    public void childHandlerInitialized(ThingHandler childHandler, Thing childThing) {
+        DevicesDTO localDeviceCaches = devicesCache;
+        if (localDeviceCaches != null) {
+            DeviceDTO device = localDeviceCaches.items.stream()
+                    .filter(d -> ((MyQDeviceHandler) childHandler).getSerialNumber().equalsIgnoreCase(d.serialNumber))
+                    .findFirst().orElse(null);
+            if (device != null) {
+                ((MyQDeviceHandler) childHandler).handleDeviceUpdate(device);
+            }
```

Comment:
The child handler can be of different type if the user made a mistake in the textual configuration. And syntactical sugar:
```suggestion
        if (localDeviceCaches != null && childHandler instanceof MyQDeviceHandler) {
            MyQDeviceHandler handler = (MyQDeviceHandler) childHandler;
            localDeviceCaches.items.stream()
                    .filter(d -> ((MyQDeviceHandler) childHandler).getSerialNumber().equalsIgnoreCase(d.serialNumber))
                    .findFirst().ifPresent(handler::handleDeviceUpdate);
```

Reply:
Did you see the syntactical sugar in my suggestion?

Reply:
I did not, thanks for the hint